### DISCUSSION TITLE
Usar TS ou não? Apenas no backend com Node? Ou em tudo?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1639] [Perguntas e Respostas]
Galera,

aqui na empresa estamos com esta dúvida... 

Às vezes me parece que usar TS no js nos fará perder toda a produtividade e velocidade que temos hoje em node / reactjs / rn... Porém, principalmente no node, as vezes falta uma maneira de seguir padrões DDD da maneira mais explicita e correta, com interfaces e etc... Já no front, não faz tanta falta assim para a gente. Aqui somos full C# / .Net Core, todo mundo bem familiarizado com POO, DDD, patterns que usam interfaces e abstrações e tipagem forte...

Mas gostaria de opiniões de vocês, TS apenas para o back com node e no front usamos a linguagem funcional e pura? Ou adotamos TS em tudo? 

Como vocês trabalham? O que acham mais vantajoso, tanto em produtividade quanto em escala?

Muito obrigado, abraços!

**Comment by ninetails** [comment]
Pessoalmente acho TS verboso, como toda linguagem com tipagem forte.

Porém acho válido tu testar já que boa parte da galera aí domina C#. Tem perdas e ganhos escrever com tipagem e isso creio que vcs já vivenciaram isso, não? Exemplo: sim, vai perder a velocidade de desenvolvimento em node porém só de pensar em tipos, ainda mais com uma galera bem acostumada a tipagem, talvez os ganhos sejam maiores na hora de desenvolver com menos bugs (pq se for usar o estrito, evitar `any` etc) vai dar pra pegar alguns erros de tipagem na hora do desenvolvimento, galera vai pensar em reestruturar numa arquitetura melhor.

Sobre usar em tudo ou não, faz um teste. Pega um projeto pequeno e tenta implementar. Tipagem não é impeditivo pra funcional, tanto que se n me engano Elm e ReasonML são tipados (se eu estiver falando merda, me perdoem e corrijam plz). Mas acho que o problema maior a ser enfrentado é muito mais o _shift_ de paradigma de OOP pra FP, coisa que teria de qualquer jeito com ou sem tipagem.

**Comment by danilosilvadev** [comment]
Eu prefiro com TS, man eu já mexi com react usando TS, node com TS e angular. Além de usar JS neles. Cara TS all the way para mim, o gasto tipando é mínimo e quando acostumado tu nem sente mais, e o ganho q tu tem. O intelisense de saber de antemão td q um objeto tem, mais a segurança antes de rodar e uma documentação viva. Eu sinceramente n entendo tanto medo de TS q a galera tem, se vc aprender bem aprendido tu gasta o q 5 min de tipagem a cada hora? 

Nos primeiros meses claro, vai ser hardcore mas ae nos próximos anos vc vê o ganho

**Comment by eliseumds** [comment]
Em tudo. Boas festas de final de ano!
